mineral-ui/mineral-ui
# [ComponentUtils] Reorganize demo to provide api examples

### Description
We wanted to update the `component-utils` docs pages to expose the API more directly while maintaining test coverage.

- add `id` to all demo examples
- make section headers linkable
- update propTypes of ComponentDoc and ComponentDocExample
- update testDemoExamples to use `id`
- update snapshots
- update unit tests

### Motivation and context
We wanted to have good content to link to while building out the Theming guide pages.

### How to test

1. `cd mineral-ui/packages/component-utils && npm start`
2. Make sure all pertinent API information is available
3. Ensure you can go directly to the anchor link for each section

### Types of changes

- Docs

### Checklist




### How does this PR make you feel?

![Alt text here](https://media.giphy.com/media/9dFvgd4ID6ne0/giphy.gif)
